core/src/AzureMcp.Core/Areas/Server/Commands/Runtime/McpRuntime.cs

```diff
@@ -53,27 +51,9 @@ public McpRuntime(
     /// <param name="request">The request context containing the tool name and parameters.</param>
     /// <param name="cancellationToken">A token to monitor for cancellation requests.</param>
     /// <returns>A result containing the output of the tool invocation.</returns>
-    public async ValueTask<CallToolResult> CallToolHandler(RequestContext<CallToolRequestParams> request, CancellationToken cancellationToken)
+    public ValueTask<CallToolResult> CallToolHandler(RequestContext<CallToolRequestParams> request, CancellationToken cancellationToken)
     {
-        using var activity = _telemetry.StartActivity(ActivityName.ToolExecuted, request?.Server?.ClientInfo);
-
-        if (request?.Params == null)
-        {
-            var content = new TextContentBlock
-            {
-                Text = "Cannot call tools with null parameters.",
-            };
-
-            activity?.SetStatus(ActivityStatusCode.Error)?.AddTag(TagName.ErrorDetails, content.Text);
-
-            return new CallToolResult
-            {
-                Content = [content],
-                IsError = true,
-            };
-        }
-
-        return await _toolLoader.CallToolHandler(request!, cancellationToken);
+        return _toolLoader.CallToolHandler(request, cancellationToken);
```

Comment:
so is it the responsibility of each tool (command) implementation to start tool-call activity?

Reply:
I would have the same question - instead of each tool loader implementation having to do activity tracking it would be easier to keep it at this top level runtime class.

Reply:
Hmm... The alternative is to start the activity and pass it to the calltoolhandler as a parameter. Which centralises this

Reply:
[Activity.Current Property](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.current?view=net-9.0)

Reply:
Yeah, it is. I have a knee jerk reaction to static things because it makes it harder to test without magic...

Reply:
Alrighty. It makes my changes a lot simpler to think of it that way.